src/d3a/models/area/event_dispatcher.py

```diff
@@ -107,17 +113,23 @@ def broadcast_balancing_market_cycle(self, **kwargs) -> None:
     def _broadcast_notification_to_single_agent(
             self, agent_area: "Area", market_type: AvailableMarketTypes,
             event_type: AreaEvent, **kwargs) -> None:
-        agent_dict = self._get_agents_for_market_type(agent_area.dispatcher, market_type)
-        for time_slot, agent in agent_dict.items():
-            if time_slot not in agent_area.get_market_instances_from_class_type(
-                    market_type):
-                # exclude past IAAs
-                continue
 
-            agent.event_listener(event_type, **kwargs)
+        if market_type == AvailableMarketTypes.FUTURE:
+            if self.future_agent:
```

Comment:
```suggestion
        if market_type == AvailableMarketTypes.FUTURE and self.future_agent:
```

Reply:
Unfortunately, this has to be like this, otherwise a future market event would be posted to the other IAAs. I had changed this maybe 5 times, because I saw the refactoring opportunity, but tests always failed. I will leave an inline comment in order to not get refactored in a future PR.